### ISSUE TITLE
request for README of pytorch-ssd-mmd-coral

### ISSUE THREAD

**Issue by chrisliu54** [VisionLearningGroup/visda-2018-public#6]
Hi, I run the `examples/ssd/train_visda.py` script after figure out the directory(using **visda18 dataset**), and I came across several errors as follows:
![image](https://user-images.githubusercontent.com/12222396/42279749-677ff0b0-7fd1-11e8-9f41-c72af47c4099.png)

After I run next time(without any change), I got the following error message:
![image](https://user-images.githubusercontent.com/12222396/42279627-fdbb4008-7fd0-11e8-8ef1-6d5e82d18b26.png)

When debugging, I found tensor operation buggy in the following code:
https://github.com/VisionLearningGroup/visda-2018-public/blob/1ef76001b1c1763295926ffe5ae5765546080052/detection/pytorch-ssd-mmd-coral/torchcv/transforms/random_flip.py#L24-L25
Specifically, when tensor pass throught the line above, the actual dimension is of 1d, not 2d.

I'm still puzzled about the dataset folder configuration. Hope the README file of the code coming out soon : )

**Comment by MInner** [comment]
As for the "Connection refused" error, install and run wisdom server in parallel, or comment lines that start it. Not sure about the latter, are you sure you downloaded our gt files and provided paths to them?

**Comment by chrisliu54** [comment]
Yep, I just download the gt files mentioned in this [page](https://github.com/VisionLearningGroup/visda-2018-public/blob/master/detection/README.md) and modified the corresponding path in the source code. I use the images extracted from `train.tar` as training data and `visda18-detection-train.txt`/`visda18-detection-test.txt` as gt boxes.

The latter problem was actually caused by dimension mismatch. Sometimes a `boxes` with only 1 dimension passed to the function. I fixed this problem by modifying the code this way:
```
xmin = w - boxes[..., 2]
xmax = w - boxes[..., 0]
```

**Comment by chrisliu54** [comment]
Well, another issue appeared when processing this line:
https://github.com/VisionLearningGroup/visda-2018-public/blob/5c5854ccd6b25428dc278c69c9c3c0469f688ccf/detection/pytorch-ssd-mmd-coral/torchcv/loss/ssd_loss.py#L63
I got the error message below:
![image](https://user-images.githubusercontent.com/12222396/42301291-ff65bc8c-8046-11e8-89fb-294a93ad51c8.png)

**Comment by heinzermch** [comment]
The whole code is written in a very sloppy way. I recommend to use [TorchCV](https://github.com/kuangliu/torchcv) directly. Which is what this project is based on anyway. Plus at least the implementation of Self-Ensemble is wrong (not that it would work for this task). I did not check the "implementation" of the other baselines, as I don't have a good understanding of them.

**Comment by MInner** [comment]
@heinzermch yep, I suspected that self-assembling code was probably wrong logically, but I would appreciate if you could point me more specifically towards what exactly is wrong with this implementation. As for the quality of code in the repo, it is pretty much torchcv SSD example with data readers for this specific dataset and somewhat different run logic since you have to train supervisely on source, unsupervisely on target, and eval on different splits of both. Pull requests are welcome :)

**Comment by krumo** [comment]
@chrisliu54 hi, did you solve this error?

**Comment by chrisliu54** [comment]
After I turned to use official repo of [TorchCV](https://github.com/kuangliu/torchcv), some weird errors has gone since then : ）
